test/tests/util.js

```diff
@@ -278,41 +284,41 @@ describe('GetFilenameFromUrl', () => {
         url: '/test/windows.txt',
         outputPath: 'c:\\foo',
         publicPath: '/test',
-        // this is weird, but it's legal-ish, and what URI parsing produces
-        expected: 'c://\\foo/windows.txt'
```

Comment:
We need ro make sure this is valid to be removed, windows paths recognize both separators to my knowledge and this teherefore will far for some folks having `/` in their paths for whatever reason: Can you elaborate on why this can be removed please ?

Reply:
https://github.com/webpack/webpack-dev-middleware/pull/276
> Where we might expect `c:\foo\bar` we instead get a URL-ized `c://\foo/bar`. Either works with the default path translation on Windows, though technically it's wrong.

If I understand correctly, the `//` is not indended to be preserved but now that it still works we didn’t bother removing it.

Reply:
Could you try with [path.normalize](https://nodejs.org/dist/latest-v10.x/docs/api/path.html#path_path_normalize_path) ?  

```js
const filename = path.posix.join(path.normalize(...))
```

Reply:
ping @michael-ciniawsky

Reply:
Paths are not URLs there must be a reason why `url-join` was used here instead. I didn't write this code so I'm not sure if 'this only handles file paths'  renders 💯 true.  I think the issue with `path.nomalize` was, that we would need to use the `path.posix` variant instead

```diff
- const filename = path.posix.join(path.normalize(...), ...)
+ const filename = path.posix.join(path.posix.normalize(...), ...)
```

But leave it for now until we know how to proceed here in general :)